cli/src/cli_util.rs

```diff
@@ -1362,6 +1375,96 @@ Then run `jj squash` to move the resolution into the conflicted commit."#,
 
         Ok(())
     }
+
+    // Retrieves pattern literals from the user's advance-branches config and parses
+    // them.
+    fn advance_branches_patterns(
+        &self,
+        setting_key: &str,
+    ) -> Result<Vec<StringPattern>, CommandError> {
+        let setting = format!("experimental-advance-branches.{setting_key}");
+        let err_msg = format!("Error parsing pattern for {setting}");
+        match self.settings.config().get_array(&setting) {
```

Comment:
nit: error handling can be simplified by `config().get::<Vec<String>>(key).optional()?`

Reply:
Looks good to me, thanks!